tools/src/main/scala/scala/scalanative/io/Classpath.scala

```diff
@@ -0,0 +1,55 @@
+package scala.scalanative.io
+
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file._
+
+import scala.collection.JavaConverters._
+
+/**
+  * A values which represents a member of a classPath.
+  */
+sealed abstract class Classpath[T <: VFile] {
+  def contents: Stream[T]
+  def close(): Unit
+}
+
+/**
+  * A trait for abstracting memory cached file-like object contained in some member of a classpath.
+  */
+trait VFile {
+  def relativePath: Path
+  def bytes: ByteBuffer
+  def uri: URI
+}
+
+final case class LocalClasspath(path: Path) extends Classpath[LazyVFile[LocalClasspath]] {
+  override def close(): Unit = ()
+
+  override def contents: Stream[LazyVFile[LocalClasspath]] = Files.walk(path,Integer.MAX_VALUE, FileVisitOption.FOLLOW_LINKS)
+    .iterator()
+    .asScala.toStream
+    .map(filePath => LazyVFile[LocalClasspath](filePath, this))
+}
+
+final case class JarClasspath(path: Path) extends Classpath[LazyVFile[JarClasspath]] {
+  val fsUri = URI.create(s"jar:${path.toUri}")
+  val options = Map("create" -> "false")
+  val fs: FileSystem = FileSystems.newFileSystem(fsUri, options.asJava)
+  val roots: Stream[Path] = fs.getRootDirectories.asScala.toStream
+
+  override def contents: Stream[LazyVFile[JarClasspath]] = {
+    roots
+      .flatMap(path => Files.walk(path, Integer.MAX_VALUE, FileVisitOption.FOLLOW_LINKS).iterator().asScala)
+      .map(path => LazyVFile[JarClasspath](path, this))
+  }
+
+  override def close(): Unit = fs.close()
+}
+
+final case class LazyVFile[T <: Classpath[LazyVFile[T]]](relativePath: Path, fs: T) extends VFile {
```

Comment:
Given that we only have LazyVirtualFiles, why even bother VirtualFile/LazyVirtualFile distinction. We don't want to prematurely abstract things.